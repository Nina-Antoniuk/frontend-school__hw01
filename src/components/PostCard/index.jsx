/* eslint-disable import/no-unresolved */
import React from 'react';
import { Link } from 'react-router-dom';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import Avatar from '@mui/material/Avatar';
import { PropTypes } from 'prop-types';
import 'animate.css';
import CardContentInfo from 'components/CardContentInfo';
import CardAction from 'components/CardAction';

const PostCard = function PostCard({
  info: { uniqueId, auth, authStats, video, hashtags, desc, comments, views },
}) {
  return (
    <Card className="animate__animated animate__pulse" sx={{ width: '100%' }}>
      <Link to={`/profile/${uniqueId}`}>
        <CardHeader
          avatar={
            <Avatar
              alt={(auth && auth.nickName) || auth.nickname}
              src={(auth && auth.avatar) || auth.avatarMedium}
              sx={{ width: 56, height: 56 }}
            />
          }
          title={(auth && auth.nickName) || auth.nickname}
        />
      </Link>
      <CardContentInfo video={video} hashtags={hashtags} desc={desc} />
      <CardAction
        auth={auth}
        authStats={authStats}
        views={views}
        comments={comments}
      />
    </Card>
  );
};

PostCard.defaultProps = {
  info: {
    uniqueId: '',
    auth: {},
    authStats: {},
    video: '',
    hashtags: [],
    desc: '',
    comments: 0,
    views: 0,
  },
};

PostCard.propTypes = {
  info: PropTypes.shape({
    uniqueId: PropTypes.string,
    auth: PropTypes.shape({
      heart: PropTypes.number,
      nickName: PropTypes.string,
      nickname: PropTypes.string,
      avatar: PropTypes.string,
      avatarMedium: PropTypes.string,
    }),
    authStats: PropTypes.shape({
      heartCount: PropTypes.number,
    }),
    video: PropTypes.string,
    hashtags: PropTypes.array,
    desc: PropTypes.string,
    comments: PropTypes.number,
    views: PropTypes.number,
  }),
};

export default PostCard;
