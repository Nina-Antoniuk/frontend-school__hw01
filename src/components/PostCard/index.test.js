import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import '@testing-library/jest-dom';
import PostCard from './index';

const fakeProps = {
  info: {
    author: {
      uniqueId: '123',
      nickname: 'Test Nickname',
      avatarMedium:
        'https://cdn.javarush.ru/images/article/a166ce03-22ed-4303-a7bf-9a52af158a7f/original.jpeg',
      signature: 'test signature',
    },
    stats: {
      commentCount: 100,
      playCount: 200,
    },
    video: 'https://www.youtube.com/watch?v=Ex2j4oIweDk',
    description: 'about unit tests',
    heart: 50,
  },
};

describe('Check PostCard component rendering', () => {
  test('with fake props', () => {
    const history = createMemoryHistory();
    const container = render(
      <Router history={history}>
        <PostCard {...fakeProps} />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });

  test('without fake props', () => {
    const history = createMemoryHistory();
    const container = render(
      <Router history={history}>
        <PostCard />
      </Router>,
    );
    expect(container).toMatchSnapshot();
  });
});

test('link path display test', () => {
  const history = createMemoryHistory();
  const { getByTestId } = render(
    <Router history={history}>
      <PostCard {...fakeProps} />
    </Router>,
  );
  const link = getByTestId('profileLink');
  expect(link.href).toContain(`/profile/${fakeProps.info.author.uniqueId}`);
});
