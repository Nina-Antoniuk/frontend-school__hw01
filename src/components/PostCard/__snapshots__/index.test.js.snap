// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check PostCard component rendering with fake props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root animate__animated animate__pulse css-tjaq38-MuiPaper-root-MuiCard-root"
      >
        <a
          class="postCard__link"
          data-testid="profileLink"
          href="/profile/123"
        >
          <div
            class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-a2xf8u-MuiAvatar-root"
                data-testid="avatar"
              >
                <img
                  alt="Test Nickname"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="https://cdn.javarush.ru/images/article/a166ce03-22ed-4303-a7bf-9a52af158a7f/original.jpeg"
                />
              </div>
            </div>
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
              >
                Test Nickname
              </span>
            </div>
          </div>
        </a>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <video
            autoplay=""
            class="video"
            controls=""
            loop=""
            preload="auto"
            src="https://www.youtube.com/watch?v=Ex2j4oIweDk"
          />
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            about unit tests
          </p>
        </div>
        <div
          class="info"
        >
          <div
            class="infoBox"
          >
            <button
              aria-label="likes"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
                data-testid="FavoriteBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            50
          </div>
          <div
            class="infoBox"
          >
            <button
              aria-label="views"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
                data-testid="RemoveRedEyeOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 6.5c3.79 0 7.17 2.13 8.82 5.5-1.65 3.37-5.02 5.5-8.82 5.5S4.83 15.37 3.18 12C4.83 8.63 8.21 6.5 12 6.5m0-2C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zm0 5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5-2.5-1.12-2.5-2.5 1.12-2.5 2.5-2.5m0-2c-2.48 0-4.5 2.02-4.5 4.5s2.02 4.5 4.5 4.5 4.5-2.02 4.5-4.5-2.02-4.5-4.5-4.5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            200
          </div>
          <div
            class="infoBox"
          >
            <button
              aria-label="comments"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
                data-testid="MessageOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 4h16v12H5.17L4 17.17V4m0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2H4zm2 10h12v2H6v-2zm0-3h12v2H6V9zm0-3h12v2H6V6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            100
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root animate__animated animate__pulse css-tjaq38-MuiPaper-root-MuiCard-root"
    >
      <a
        class="postCard__link"
        data-testid="profileLink"
        href="/profile/123"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-a2xf8u-MuiAvatar-root"
              data-testid="avatar"
            >
              <img
                alt="Test Nickname"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="https://cdn.javarush.ru/images/article/a166ce03-22ed-4303-a7bf-9a52af158a7f/original.jpeg"
              />
            </div>
          </div>
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
            >
              Test Nickname
            </span>
          </div>
        </div>
      </a>
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <video
          autoplay=""
          class="video"
          controls=""
          loop=""
          preload="auto"
          src="https://www.youtube.com/watch?v=Ex2j4oIweDk"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          about unit tests
        </p>
      </div>
      <div
        class="info"
      >
        <div
          class="infoBox"
        >
          <button
            aria-label="likes"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
              data-testid="FavoriteBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          50
        </div>
        <div
          class="infoBox"
        >
          <button
            aria-label="views"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
              data-testid="RemoveRedEyeOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 6.5c3.79 0 7.17 2.13 8.82 5.5-1.65 3.37-5.02 5.5-8.82 5.5S4.83 15.37 3.18 12C4.83 8.63 8.21 6.5 12 6.5m0-2C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zm0 5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5-2.5-1.12-2.5-2.5 1.12-2.5 2.5-2.5m0-2c-2.48 0-4.5 2.02-4.5 4.5s2.02 4.5 4.5 4.5 4.5-2.02 4.5-4.5-2.02-4.5-4.5-4.5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          200
        </div>
        <div
          class="infoBox"
        >
          <button
            aria-label="comments"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
              data-testid="MessageOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 4h16v12H5.17L4 17.17V4m0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2H4zm2 10h12v2H6v-2zm0-3h12v2H6V9zm0-3h12v2H6V6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          100
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Check PostCard component rendering without fake props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root animate__animated animate__pulse css-tjaq38-MuiPaper-root-MuiCard-root"
      >
        <a
          class="postCard__link"
          data-testid="profileLink"
          href="/profile/undefined"
        >
          <div
            class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular css-a2xf8u-MuiAvatar-root"
                data-testid="avatar"
              >
                <img
                  alt="placeholder for stranger"
                  class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                  src="https://via.placeholder.com/56x56"
                />
              </div>
            </div>
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
              >
                stranger
              </span>
            </div>
          </div>
        </a>
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <img
            alt="placeholder"
            class="placeholder"
            src="https://via.placeholder.com/250x300.png?text=video+is+not+available"
          />
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          />
        </div>
        <div
          class="info"
        >
          <div
            class="infoBox"
          >
            <button
              aria-label="likes"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
                data-testid="FavoriteBorderIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            0
          </div>
          <div
            class="infoBox"
          >
            <button
              aria-label="views"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
                data-testid="RemoveRedEyeOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 6.5c3.79 0 7.17 2.13 8.82 5.5-1.65 3.37-5.02 5.5-8.82 5.5S4.83 15.37 3.18 12C4.83 8.63 8.21 6.5 12 6.5m0-2C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zm0 5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5-2.5-1.12-2.5-2.5 1.12-2.5 2.5-2.5m0-2c-2.48 0-4.5 2.02-4.5 4.5s2.02 4.5 4.5 4.5 4.5-2.02 4.5-4.5-2.02-4.5-4.5-4.5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            0
          </div>
          <div
            class="infoBox"
          >
            <button
              aria-label="comments"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
                data-testid="MessageOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M4 4h16v12H5.17L4 17.17V4m0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2H4zm2 10h12v2H6v-2zm0-3h12v2H6V9zm0-3h12v2H6V6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            0
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root animate__animated animate__pulse css-tjaq38-MuiPaper-root-MuiCard-root"
    >
      <a
        class="postCard__link"
        data-testid="profileLink"
        href="/profile/undefined"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-a2xf8u-MuiAvatar-root"
              data-testid="avatar"
            >
              <img
                alt="placeholder for stranger"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="https://via.placeholder.com/56x56"
              />
            </div>
          </div>
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
            >
              stranger
            </span>
          </div>
        </div>
      </a>
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <img
          alt="placeholder"
          class="placeholder"
          src="https://via.placeholder.com/250x300.png?text=video+is+not+available"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        />
      </div>
      <div
        class="info"
      >
        <div
          class="infoBox"
        >
          <button
            aria-label="likes"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
              data-testid="FavoriteBorderIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          0
        </div>
        <div
          class="infoBox"
        >
          <button
            aria-label="views"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
              data-testid="RemoveRedEyeOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 6.5c3.79 0 7.17 2.13 8.82 5.5-1.65 3.37-5.02 5.5-8.82 5.5S4.83 15.37 3.18 12C4.83 8.63 8.21 6.5 12 6.5m0-2C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zm0 5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5-2.5-1.12-2.5-2.5 1.12-2.5 2.5-2.5m0-2c-2.48 0-4.5 2.02-4.5 4.5s2.02 4.5 4.5 4.5 4.5-2.02 4.5-4.5-2.02-4.5-4.5-4.5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          0
        </div>
        <div
          class="infoBox"
        >
          <button
            aria-label="comments"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-u0nhjn-MuiSvgIcon-root"
              data-testid="MessageOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 4h16v12H5.17L4 17.17V4m0-2c-1.1 0-1.99.9-1.99 2L2 22l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2H4zm2 10h12v2H6v-2zm0-3h12v2H6V9zm0-3h12v2H6V6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          0
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
